src/viser/_gui_handles.py

```diff
@@ -298,6 +305,17 @@ def disabled(self, disabled: bool) -> None:
         assert not disabled, "Button groups cannot be disabled."
 
 
+@dataclasses.dataclass
+class GuiNotificationHandle:
+    """Handle for a notification in our visualizer."""
+
+    gui_api: GuiApi
+    notification: NotificationMessage
+
+    def show(self):
```

Comment:
Do you think it's very useful to have a `.show()`, as opposed to just showing the notification immediately?

A `.close()` might be more useful to remove just a single notification, as well as (not necessarily now) updating the content / properties of existing notifications

Reply:
Ok that sounds good! I was trying to replicate what I thought was like the other gui things but realize now isn't very similar.

Reply:
Hmm, a workflow like this could be nice: https://v4.mantine.dev/others/notifications/#update-notification ... like an open/close/update.

Reply:
Update functionality! You cannot change the actual `type` of the notification but you can get rid of the `loading` swirl.

https://github.com/nerfstudio-project/viser/assets/42229107/2eb6ab85-c31e-40fc-a925-bf8300d4eaf7

Reply:
@chungmin99 While using notifications for nerfstudio, I find that `notifications.update(...)` is super helpful. However, I think some more granular features are pretty important to include in `update`. For example, I might want to change the `loading` state, or add a close button, which are features that were originally abstracted away after the different types (controlled, persistent, and timed) are added. Maybe instead of splitting notifications into these "types," we still include all the granular features but just have default values for them. What do you think?

Reply:
> Maybe instead of splitting notifications into these "types," we still include all the granular features but just have default values for them.

This makes sense to me! I see the appeal of the discrete types, but it's also nice to maximize flexibility here.

For updating individual attributes, matching the setter/getter-style properties used by other handle objects could be nice for being idiomatic/consistent: 
https://github.com/nerfstudio-project/viser/blob/d56be1ad1602739594cb30307cd61599a696d588/src/viser/_gui_handles.py#L164-L178